### PR TITLE
storage: exclude os from the default probe set

### PR DESCRIPTION
**Pull request by dbungert**
Exclude 'os' from the default storage probes.
Add a 'defaults' keyword, which can be used in a set with 'os' to get both the
defaults and the 'os' result.

### CODE REVIEW

File: probert/storage.py [hunk 1]
```diff
@@ -147,44 +154,49 @@ class Storage():
...
-        'zfs': zfs.probe
+        'bcache': Probe(bcache.probe),
+        'blockdev': Probe(blockdev_probe),
+        'dasd': Probe(dasd.probe),
+        'dmcrypt': Probe(dmcrypt.probe),
+        'filesystem': Probe(filesystem.probe),
+        'lvm': Probe(lvm.probe),
+        'mount': Probe(mount.probe),
+        'multipath': Probe(multipath.probe),
+        'os': Probe(os.probe, False),
```
Comment (mwhudson) [inline probert/storage.py:165]: Use a keyword argument here?

File: probert/storage.py [hunk 2]
```diff
@@ -147,44 +154,49 @@ class Storage():
...
 
     def __init__(self, results={}):
         self.results = results
         self.context = pyudev.Context()
 
-    def _get_probe_types(self):
-        return {ptype for ptype, pfunc in self.probe_map.items() if pfunc}
+    def _get_probe_types(self, all=False):
+        return {ptype for ptype, probe in self.probe_map.items()
+                if probe.pfunc and (probe.in_default_set or all)}
```
Comment (mwhudson) [inline probert/storage.py:176]: Is there any way probe.pfunc can be non-truthy here? I never understood this bit of the code :)

**Review by mwhudson — APPROVED** [review]
Needs trivial rebase ofc.

File: probert/storage.py (hunk 2 above)
Comment (dbungert) [reply probert/storage.py:176]: As far as I can tell it's always going to be true.  This reads like a C-ism, and if we wanted this then `getattr(foo, 'pfunc')` is probably what was desired.  I'm just removing the pfunc reference.

**Review by mwhudson — APPROVED** [review]

**Pull request merged**